### ISSUE TITLE
New problem in fontforge-cygwin_2014_01_04: Generate('a.ttf') takes 2x as much memory as the version of 2012

### ISSUE THREAD

**Issue by ilyaza** [fontforge/fontforge#1232]
For my files, this does not matter: my workflow separates Generate() into a separate call
    fontforge -lang ff -c "Load($1); Generate($2)"
and for me, the question is about using 500MB of memory vs 1100MB of memory; not a problem with this machine.  However, people with larger runs may get into problems on 32bit machines, more complicated fonts, and/or more complicated scripts.

**Comment by serval2412** [comment]
Just for the record, I submitted several patches concerning ressource leaks (memory, file descriptor) these last days.
However https://scan.coverity.com/ shows there are still a bit less of 200 leaks remaining.
Of course, there may be some false positives but it worths it to take it a look.
There's https://github.com/fontforge/fontforge/pull/3623 which may tackle several leaks.

**Comment by davelab6** [comment]
Thanks Julien!!!

**Comment by jtanx** [comment]
So I was still curious about this one, but I figured it out in the end.

Usage drops from ~1.1GB->640MB when using floats over doubles. The rest of the difference came from using George's custom chunkalloc allocator over calloc (which we always use now) - bringing it down to around ~560MB.

We've added some fields onto the data structures since then, which has also skewed it up slightly.

A nice breakdown (note this is 64-bit with double precision reals):
![image](https://user-images.githubusercontent.com/5137410/81061666-199e1900-8f18-11ea-89e5-6d5f4ddc8289.png)
![image](https://user-images.githubusercontent.com/5137410/81061703-2a4e8f00-8f18-11ea-8edf-1b4f57f66587.png)

Most of it comes from Spline/SplinePoint allocations (no surprises there).

Bottom line is it's nothing we're going to change today, memory's relatively cheap so I doubt it's much of a problem anyway. There's even 64 bit builds for Windows if you so desire.
